sequential/seq2pat.py

```diff
@@ -283,12 +294,36 @@ class Seq2Pat:
         sequences but no constraints are used to run the mining efficiently and practically.
         Power users can choose to drop this constraint by setting it to be None or increase the maximum span
         as the system has resources to support.
+    batch_size: Optional[int]
+        When batch_size is set, program runs Seq2Pat on batches of sequences instead of on the entire set of
+        sequences for improving scalability. Each batch contains `batch_size` sequences as a random sample of
+        entire set. A mining task will run on each batch with a reduced minimum row count (min_frequency) threshold.
+        Please refer to description of discount_factor parameter for how min_frequency is reduced.
+        Resulted patterns will be aggregated from the mining results of each batch by calculating the sum of
+        the occurrences. Finally the original minimum row count threshold is applied to the patterns after aggregation.
+    n_jobs: int
+        n_jobs defines the number of processes (n_jobs=2 by default) that are used when mining tasks are applied
+        on batches in parallel. If -1 all CPUs are used. If -2, all CPUs but one are used.
+    seed: int
+        Random seed to make sequences uniformly distributed among batches.
+    discount_factor: float
+        A discount factor is used to reduce the minimum row count (min_frequency) threshold when Seq2Pat is applied
+        on a batch. When min_frequency is a integer, mining task can only be run on the entire set.
```

Comment:
aN integer

Reply:
Not sure why can't we do batching with Integer min_freq.. cannot we find float version of a given min_freq_integer by looking at min_freq_integer and num_total_rows?

Reply:
My immediate thought is to explicitly require it to be a float, then there would be less conversions or rounding between integer and ratio thresholds. But since the default seq2pat usage should support integer threshold, it might be better to allow integer in batch mode. I can add the automated conversion to it.